### PR TITLE
Avoid escaping of characters other than "

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

replaces #20
refs https://github.com/matomo-org/matomo/issues/19899

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: src/IniReader.php [hunk 1]
```diff
@@ -284,7 +284,15 @@ private function readWithAlternativeImplementation($ini)
...
-                $value = trim($value, "'\"");
+                if (preg_match('/^"(.*)"$/', $value)) {
+                    $value = preg_replace('/^"(.*)"$/', '$1', $value);
+                    $value = str_replace('\"', '"', $value);
+                } elseif (preg_match("/^'(.*)'$/", $value)) {
+                    $value = preg_replace("/^'(.*)'$/", '$1', $value);
+                    $value = str_replace("\'", "'", $value);
+                } else {
+                    $value = trim($value, "'\"");
+                }
```
Comment (sgiehl) [inline src/IniReader.php:295]: This actually fixes some difference between this reader implementation and `parse_ini_string`.
Values like `key = "test \" test"` or `key = "' test"` resulted in different results. This should be covered by the additional tests

**Review by peterhashair — APPROVED** [review]
Looks good to me.

* [x] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [x] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [x] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [x] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [x] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [x] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [x] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [x] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)

**Comment by justinvelluppillai** [comment]
@peterhashair @sgiehl any reason this isn't merged if it is fully tested and reviewed?

**Comment by peterhashair** [comment]
@justinvelluppillai not sure if there is there any special process for this repo, but looks good to me to do a merge.
